modules/add-contextual-data/context-info-db.c

```diff
@@ -119,14 +146,28 @@ _record_free(gpointer p)
   contextual_data_record_clean(rec);
 }
 
-static void
-_new(ContextInfoDB *self)
+static gboolean
+_strcase_eq(gconstpointer a, gconstpointer b)
+{
+  return _g_strcasecmp(a,b) == 0;
+}
+
+static guint
+_strcase_hash(gconstpointer value)
+{
+  gchar *upper_case_value = _str_case_insensitive_dup((const gchar *)value);
+  guint hash = g_str_hash(upper_case_value);
+  g_free(upper_case_value);
+  return hash;
+}
```

Comment:
hmm. I would consider doing this without the memory allocation if possible.

Reply:
yes, agree(and it is possible, maybe we need to implement a custom hash func).